tests/vm_deletion_protection_test.go

```diff
@@ -0,0 +1,83 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/ssp-operator/tests/env"
+)
+
+var _ = Describe("VM delete protection", func() {
+	const deleteProtectionLabel = "kubevirt.io/vm-delete-protection"
+
+	var vm *kubevirtv1.VirtualMachine
+
+	var createVMWithDeleteProtection = func(protected string) {
+		vmName := fmt.Sprintf("testvmi-%v", rand.String(10))
+		vmi := NewMinimalVMIWithNS(strategy.GetNamespace(), vmName)
+		vm = NewVirtualMachine(vmi)
+
+		vm.Labels = make(map[string]string)
+		vm.Labels[deleteProtectionLabel] = protected
+		eventuallyCreateVm(vm)
+	}
+
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	AfterEach(func() {
+		err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
+		if !errors.IsNotFound(err) {
+			Eventually(func(g Gomega) error {
+				err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
+				if err != nil {
+					return err
+				}
+
+				vm.Labels[deleteProtectionLabel] = "False"
+				return apiClient.Update(ctx, vm)
+			}, env.ShortTimeout(), time.Second).Should(Succeed())
+		}
+	})
+
+	It("should not allow to delete a VM if the protection is enabled", func() {
+		createVMWithDeleteProtection("True")
+
+		err := apiClient.Delete(ctx, vm)
+
+		Expect(err).To(
+			MatchError(ContainSubstring(fmt.Sprintf("VirtualMachine %v cannot be deleted, remove delete protection", vm.Name))))
```

Comment:
ContainSubstring() already takes format string parameter, so no need to call fmt.Sprintf()

Reply:
Ah, you right. Thanks. Done.